packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/InAppPurchasesModule.java

```diff
@@ -11,9 +11,9 @@
 import org.unimodules.core.interfaces.ExpoMethod;
 import org.unimodules.core.interfaces.ActivityProvider;
 import org.unimodules.core.interfaces.services.EventEmitter;
-import org.unimodules.core.interfaces.RegistryLifecycleListener;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
 
-public class InAppPurchasesModule extends ExportedModule implements RegistryLifecycleListener {
+public class InAppPurchasesModule extends ExportedModule implements ModuleRegistryConsumer {
```

Comment:
How are you testing those changes? If you're testing the module using `apps/expo-payments` I must warn you—it should be configured differently:

- adding `name` and `version` to `apps/expo-payments/package.json` would make it compatible with yarn workspaces
- instead of depending on `expokit` from NPM (exact version downloads tgz from NPM), we could depend on yarn workspaces and link the `expokit` from `expokit-npm-package`. (The project should be doing it already on Android 🤔)

https://github.com/expo/expo/blob/122ae38b06741b11a02a72deea1748a9fbc8e223/apps/expo-payments/android/app/build.gradle#L186-L189

- On iOS Podfile uses ExpoKit from Git (not local ExpoKit.podspec)

    https://github.com/expo/expo/blob/122ae38b06741b11a02a72deea1748a9fbc8e223/apps/expo-payments/ios/Podfile#L5-L7

    which could be changed to:

    ```rb
      pod 'ExpoKit',
        :path => "../../../",
    ```

Reply:
This change would probably make sense on `sdk-33` branch from where we publish sdk33-compatible unimodules, `master` is a place for upcoming SDK. 🙂

Reply:
So merge this whole PR into the `sdk-33` branch? Or just this commit: https://github.com/expo/expo/pull/4723/commits/68ef7541f0708be42ec8f703fbfd2152b427fc74.

I imagine the README/package.json should be updated in both right? But also the IAP package doesn't exist at all in that branch since I merged that PR into master. Do you want me to cherrypick that too?

Reply:
Yeah, I was originally testing using `apps/expo-payments` but ran into some issues. After publishing the package, I've been using an ejected Expokit app as well. Hence why I want to introduce these changes to the Android module.